andela/travel_tool_back
# Ft etl travel calendar 161781854

#### What does this PR do?
This PR performs the ETL task on the travel calendar data from the stakeholders

#### Description of Task to be completed?
- Clean up the google sheets data.
- Transform data to JSON format.
- Use the data to create the user objects by fetching their data from the ANDELA API and the Bamboo HR API.
- Create the users and their respective managers on the users table.
- Create the requests in the requests table.
- Create the trip for the users on the trips table.
- Create the ticket details in the travel checklist submissions table.

#### How should this be manually tested?
1. Clone the repository `https://github.com/andela/travel_tool_back`

2. Add these keys in your .env file:
ANDELA_PROD_API=https://api-prod.andela.com/api/v1
BAMBOOHR_API=https://925f384d6f711e345a2a3404923ba13801ceb974:x@api.bamboohr.com/api/gateway.php/andela/v1/employees/{bambooHRId}?fields=id,displayName,firstName,lastName,jobTitle,department,division,location,workEmail,supervisor,supervisorEId
DEFAULT_SUPER_USER_ID=1234abcd
USER_JWT={put your production jwt token here}

3. Change to the travel_tool directory on your editor and run the following commands:
* git checkout `ft-etl-travel-calendar-161781854` in travel_tool_back 
* `yarn db:rollback && yarn db:migrate && yarn db:seed` 
* `node migrator/migrator` 
>For `make` commands:
* `make rollback`
* `make migrate`
* `make seed`
* `node migrator/migrator.js`

4. Check your database for migrated details in Users, UserRoles, Requests, Approvals, Trips, Checklist Submission

5. Check the failed migrations in the `usersNotFoundProd.txt` and `usersNotFoundBamboo.txt`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#161781854](https://www.pivotaltracker.com/story/show/161781854)

#### Screenshots (if appropriate)

#### Questions:
